mountpoint-s3/src/data_cache/disk_data_cache.rs

```diff
@@ -6,24 +6,45 @@ use std::ops::RangeBounds;
 use std::path::PathBuf;
 
 use base64ct::{Base64UrlUnpadded, Encoding};
-use bytes::{BufMut, BytesMut};
-use mountpoint_s3_crt::checksums::crc32c;
+use bytes::{BufMut, Bytes, BytesMut};
+use serde::{Deserialize, Serialize};
 use tracing::{error, trace, warn};
 
 use crate::data_cache::DataCacheError;
+use crate::serde::SerializableCrc32c;
 
 use super::{BlockIndex, CacheKey, ChecksummedBytes, DataCache, DataCacheResult};
 
 /// On-disk implementation of [DataCache].
 ///
-/// TODO: Store checksums on disk, reconstruct as [ChecksummedBytes] using same checksum avoiding recomputation.
-///
 /// TODO: Store additional metadata with each block such as expected S3 key, ETag, etc..
 pub struct DiskDataCache {
     block_size: u64,
     cache_directory: PathBuf,
 }
 
+/// Represents a fixed-size chunk of data that can be serialized.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct DataBlock {
+    checksum: SerializableCrc32c,
```

Comment:
Do we need `SerializableCrc32c`? This struct is already kind of a "serializable version" of `ChecksummedBytes`, maybe we just store the u32 for the checksum here and rely on derive.